linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -5247,7 +5241,18 @@ private[optimizer] object OptimizerCore {
         This()(tpe.base)
 
       case ReplaceWithOtherLocalDef(localDef) =>
-        newReplacementInternal(localDef.replacement)
+        /* A previous version would push down the `tpe` of this `LocalDef` to
+         * use for the replacement. While that creates trees with narrower types,
+         * it also creates inconsistent trees:
+         * - This() not typed as the enclosing class.
+         * - VarRef not typed as the corresponding VarDef / ParamDef.
+         *
+         * Type based optimizations happen (mainly) in the optimizer so
+         * consistent downstream types are more important than narrower types;
+         * notably because it allows us to run the ClassDefChecker after the
+         * optimizer.
+         */
+        localDef.newReplacement
```

Comment:
Does this change result in any different in the produced `.js` for the test suite?

Reply:
Nope. They are byte by byte identical.